openrouteservice/src/main/java/org/heigit/ors/fastisochrones/Contour.java

```diff
@@ -0,0 +1,386 @@
+
+        List<Coordinate> points = new ArrayList<>(1 / 20 * coordinates.size());
+        PointItemVisitor visitor = new PointItemVisitor(0, 0, defaultVisitorThreshold);
+        Quadtree qtree = new Quadtree();
+        Envelope searchEnv = new Envelope();
+        TreeSet<Coordinate> treeSet = new TreeSet<>();
+
+        while (j < coordinates.size()) {
+            double latitude = coordinates.get(j).y;
+            double longitude = coordinates.get(j).x;
+            j++;
+            addPoint(visitor, points, qtree, searchEnv, treeSet, longitude, latitude, defaultSearchWidth, defaulPointWidth, true);
+        }
+
+        GeometryFactory geomFactory = new GeometryFactory();
+        int size = points.size();
+        Geometry[] geometries = new Geometry[size];
+        int g = 0;
+        for (int i = 0; i < size; i++)
+            geometries[g++] = geomFactory.createPoint(points.get(i));
+        GeometryCollection treePoints = new GeometryCollection(geometries, geomFactory);
+
+        ConcaveHull ch = new ConcaveHull(treePoints, CONCAVE_HULL_THRESHOLD, false);
+        return ch.getConcaveHull();
+    }
+
+    private Boolean addPoint(PointItemVisitor visitor, List<Coordinate> points, Quadtree tree, Envelope searchEnv, Set<Coordinate> treeSet, double lon, double lat, double searchWidth, double pointWidth, boolean checkNeighbours) {
+        if (checkNeighbours) {
+            visitor.setPoint(lon, lat);
+            searchEnv.init(lon - searchWidth, lon + searchWidth, lat - searchWidth, lat + searchWidth);
+            tree.query(searchEnv, visitor);
+            if (!visitor.isNeighbourFound()) {
+                Coordinate p = new Coordinate(lon, lat);
+
+                if (!treeSet.contains(p)) {
+                    Envelope env = new Envelope(lon - pointWidth, lon + pointWidth, lat - pointWidth, lat + pointWidth);
+                    tree.insert(env, p);
+                    points.add(p);
+                    treeSet.add(p);
+
+                    return true;
+                }
+            }
+        } else {
+            Coordinate p = new Coordinate(lon, lat);
+            if (!treeSet.contains(p)) {
+                Envelope env = new Envelope(lon - pointWidth, lon + pointWidth, lat - pointWidth, lat + pointWidth);
+                tree.insert(env, p);
+                points.add(p);
+                treeSet.add(p);
+
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private IntObjectMap<IntHashSet> identifySuperCells(IntSet cellIds, int hierarchyLevel, boolean isPrimary) {
+        //Account for the subcell division in InertialFlow final step
+        int maxId = createMaxId(cellIds);
+
+        IntHashSet visitedCells = new IntHashSet();
+        IntObjectMap<IntHashSet> superCells = new IntObjectHashMap<>();
+        List<Integer> orderedCellIds = Arrays.stream(cellIds.toArray()).boxed().collect(Collectors.toList());
+        Collections.sort(orderedCellIds);
```

Comment:
Sorted the stream directly by putting sorted() between boxed() and collect() instead of using Collections.sort(). https://github.com/GIScience/openrouteservice/commit/e686de44d05f72f304d5bea045c5f514370d58b0#diff-931f1f8c6693ac81fccb90df1537fafd